devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -291,7 +291,10 @@ def handleBatteryEvents(descMap) {
 			batteryValue = Math.round(rawValue / 2)
 		}
 
-		if (batteryValue != null) {
+		// Some special logic for the IKEA Dimmer:
+		// Only report the event if we have a legal battery reading, and the difference between this reading and the last reading isn't > 20%.
+		// Sometimes we receive spurious low battery events.
+		if (batteryValue != null && !(isIkeaDimmer() && device.currentValue("battery") && (device.currentValue("battery") as int) - batteryValue > 20)) {
```

Comment:
I'm not familiar with the values typically reported from this device but for other devices we've implemented an algorithm where the device needs to report a new value 2 (or more) times before it would be considered valid. That way if the value actually changes it does get reported eventually.

Reply:
Based on the issue it looks like these reports come too frequently to rely on that.

Reply:
what if we kept a running average of the last, say, ten values?

Reply:
A running average could work. It seems that the battery values fluctuate based on the usage at the time. I seem to recall that the device only responds to battery events when the device is being used.

Some of the reported values were: 37, 11, 37, 11, 37, 11, 17, 8, 11, 0, 11, 17, 11, 17.

I've never actually experienced reporting as frequent as certification has.